platforms/cpus/rs_virgo.repl

```diff
@@ -44,19 +40,72 @@ spi: Memory.MappedMemory @ sysbus 0xA0800000
     size: 0x8000
 wdt: Memory.MappedMemory @ sysbus 0xA0120000
     size: 0x4000
-i2c: Memory.MappedMemory @ sysbus 0xA0410000
-    size: 0x10000
-
-
+    
 // implemented
 gpt: Timers.ATCPIT100 @ sysbus 0xA0520000
 
 gpio: GPIOPort.ATCGPIO100 @ sysbus 0xA0510000
     cfgGpioNum: 8 
+    0 -> pad@16 
+    1 -> pad@17
+    2 -> pad@26
+    3 -> pad@27
+    4 -> pad@28
+    5 -> pad@29
+    6 -> pad@30
+    7 -> pad@31
 
+FPGA_GPIO:GPIOPort.ATCGPIO100 @ sysbus 0xA0540000
+    0 -> pad@32
+    1 -> pad@33
+    2 -> pad@34
+    3 -> pad@35
+    4 -> pad@36
+    5 -> pad@37
+    6 -> pad@38
+    7 -> pad@39
+    8 -> pad@40
+    9 -> pad@41
+    10-> pad@42
+    11-> pad@43
+    12-> pad@44
+    13-> pad@45
+    14-> pad@46
+    15-> pad@47
+        
 uart0: UART.ATCUART100 @ sysbus 0xA0420000
     //IRQ -> plic@5
     //clockFrequency : 66000000
-
+    
 dma: DMA.ATCDMAC100 @ sysbus 0xA0600000
-    IRQ -> scu@27
\ No newline at end of file
+   // IRQ -> scu@27 
+   
+pad: Miscellaneous.Virgo_pad @ sysbus 0xA4100000   
+//main function I/Os are in range 16-31
+    16 ->gpio@0
+    17 ->gpio@1
+    26 ->gpio@2
+    27 ->gpio@3
+    28 ->gpio@4
+    29 ->gpio@5
+    30 ->gpio@6
+    31 ->gpio@7
+//fpga_gpio I/Os are in range 32-47
```

Comment:
Probably you can increase this comment by mentioning that FPGA GPIO mode is currently supported by using an ATCGPIO100.